beacon-chain/sync/initial-sync/blocks_fetcher_utils.go

```diff
@@ -23,14 +24,14 @@ func (f *blocksFetcher) nonSkippedSlotAfter(ctx context.Context, slot uint64) (u
 	defer span.End()
 
 	headEpoch := helpers.SlotToEpoch(f.headFetcher.HeadSlot())
-	finalizedEpoch, peers := f.p2p.Peers().BestFinalized(params.BeaconConfig().MaxPeersToSync, headEpoch)
+	targetEpoch, peers := f.p2p.Peers().BestNonFinalized(flags.Get().MinimumSyncPeers, headEpoch)
```

Comment:
doesn't this also need to check the mode before deciding which method to use

Reply:
I believe it will be the most safe way to do it -- will update.